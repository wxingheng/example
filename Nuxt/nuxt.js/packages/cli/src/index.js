import * as _commands from './commands'
import * as _imports from './imports'

export const commands = _commands
export const imports = _imports

export { default as NuxtCommand } from './command'
export { default as setup } from './setup'
export { default as run } from './run'
export { loadNuxtConfig } from './utils'
