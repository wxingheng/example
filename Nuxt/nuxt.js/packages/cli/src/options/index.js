export { default as common } from './common'
export { default as server } from './server'
export { default as locking } from './locking'
