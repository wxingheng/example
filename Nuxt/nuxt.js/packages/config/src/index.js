export { defaultNuxtConfigFile, getDefaultNuxtConfig } from './config'
export { getNuxtConfig } from './options'
