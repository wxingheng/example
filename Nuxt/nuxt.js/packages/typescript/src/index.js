import { readJSON, writeJSON } from 'fs-extra'
import { register } from 'ts-node'
import defaultsDeep from 'lodash/defaultsDeep'

export const defaultTsJsonConfig = {
  compilerOptions: {
    target: 'esnext',
    module: 'esnext',
    moduleResolution: 'node',
    lib: [
      'esnext',
      'esnext.asynciterable',
      'dom'
    ],
    esModuleInterop: true,
    experimentalDecorators: true,
    allowJs: true,
    sourceMap: true,
    strict: true,
    noImplicitAny: false,
    noEmit: true,
    baseUrl: '.',
    paths: {
      '~/*': [
        './*'
      ],
      '@/*': [
        './*'
      ]
    },
    types: [
      '@types/node',
      '@nuxt/vue-app'
    ]
  }
}

let _setup = false

export async function setup(tsConfigPath, options = {}) {
  if (_setup) {
    return
  }
  _setup = true

  const config = await readJSON(tsConfigPath)
  await writeJSON(tsConfigPath, defaultsDeep(config, defaultTsJsonConfig), { spaces: 2 })

  // https://github.com/TypeStrong/ts-node
  register({
    project: tsConfigPath,
    compilerOptions: {
      module: 'commonjs'
    },
    ...options
  })
}
