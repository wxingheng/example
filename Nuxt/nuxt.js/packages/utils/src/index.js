export * from './context'
export * from './lang'
export * from './locking'
export * from './resolve'
export * from './route'
export * from './serialize'
export * from './task'
export * from './timer'
