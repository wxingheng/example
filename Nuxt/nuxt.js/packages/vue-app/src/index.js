import path from 'path'
import { dependencies } from '../package.json'

export const template = {
  dependencies,
  dir: path.join(__dirname, '..', 'template'),
  files: [
    'App.js',
    'client.js',
    'index.js',
    'middleware.js',
    'router.js',
    'server.js',
    'utils.js',
    'empty.js',
    'components/nuxt-error.vue',
    'components/nuxt-loading.vue',
    'components/nuxt-child.js',
    'components/nuxt-link.server.js',
    'components/nuxt-link.client.js',
    'components/nuxt.js',
    'components/no-ssr.js',
    'views/app.template.html',
    'views/error.html'
  ]
}
