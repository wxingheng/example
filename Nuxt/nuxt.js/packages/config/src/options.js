import path from 'path'
import fs from 'fs'
import defaultsDeep from 'lodash/defaultsDeep'
import defaults from 'lodash/defaults'
import pick from 'lodash/pick'
import isObject from 'lodash/isObject'
import uniq from 'lodash/uniq'
import consola from 'consola'
import { guardDir, isNonEmptyString, isPureObject, isUrl, getMainModule } from '@nuxt/utils'
import { defaultNuxtConfigFile, getDefaultNuxtConfig } from './config'

export function getNuxtConfig(_options) {
  // Prevent duplicate calls
  if (_options.__normalized__) {
    return _options
  }

  // Clone options to prevent unwanted side-effects
  const options = Object.assign({}, _options)
  options.__normalized__ = true

  // Normalize options
  if (options.loading === true) {
    delete options.loading
  }
  if (
    options.router &&
    options.router.middleware &&
    !Array.isArray(options.router.middleware)
  ) {
    options.router.middleware = [options.router.middleware]
  }
  if (options.router && typeof options.router.base === 'string') {
    options._routerBaseSpecified = true
  }
  if (typeof options.transition === 'string') {
    options.transition = { name: options.transition }
  }
  if (typeof options.layoutTransition === 'string') {
    options.layoutTransition = { name: options.layoutTransition }
  }
  if (typeof options.extensions === 'string') {
    options.extensions = [options.extensions]
  }

  options.globalName = (isNonEmptyString(options.globalName) && /^[a-zA-Z]+$/.test(options.globalName))
    ? options.globalName.toLowerCase()
    : `nuxt`

  // Resolve rootDir
  options.rootDir = isNonEmptyString(options.rootDir) ? path.resolve(options.rootDir) : process.cwd()

  // Apply defaults by ${buildDir}/dist/build.config.js
  // TODO: Unsafe operation.
  // const buildDir = options.buildDir || defaults.buildDir
  // const buildConfig = resolve(options.rootDir, buildDir, 'build.config.js')
  // if (existsSync(buildConfig)) {
  //   defaultsDeep(options, require(buildConfig))
  // }

  // Apply defaults
  const nuxtConfig = getDefaultNuxtConfig()

  nuxtConfig.build._publicPath = nuxtConfig.build.publicPath

  // Fall back to default if publicPath is falsy
  if (options.build && !options.build.publicPath) {
    options.build.publicPath = undefined
  }

  defaultsDeep(options, nuxtConfig)

  // Check srcDir and generate.dir existence
  const hasSrcDir = isNonEmptyString(options.srcDir)
  const hasGenerateDir = isNonEmptyString(options.generate.dir)

  // Resolve srcDir
  options.srcDir = hasSrcDir
    ? path.resolve(options.rootDir, options.srcDir)
    : options.rootDir

  // Resolve buildDir
  options.buildDir = path.resolve(options.rootDir, options.buildDir)

  // Default value for _nuxtConfigFile
  if (!options._nuxtConfigFile) {
    options._nuxtConfigFile = path.resolve(options.rootDir, `${defaultNuxtConfigFile}.js`)
  }

  // Watch for _nuxtConfigFile changes
  options.watch.push(options._nuxtConfigFile)

  // Protect rootDir against buildDir
  guardDir(options, 'rootDir', 'buildDir')

  if (hasGenerateDir) {
    // Resolve generate.dir
    options.generate.dir = path.resolve(options.rootDir, options.generate.dir)

    // Protect rootDir against buildDir
    guardDir(options, 'rootDir', 'generate.dir')
  }

  if (hasSrcDir) {
    // Protect srcDir against buildDir
    guardDir(options, 'srcDir', 'buildDir')

    if (hasGenerateDir) {
      // Protect srcDir against generate.dir
      guardDir(options, 'srcDir', 'generate.dir')
    }
  }

  // Populate modulesDir
  options.modulesDir = uniq(
    getMainModule().paths.concat(
      [].concat(options.modulesDir).map(dir => path.resolve(options.rootDir, dir))
    )
  )

  const mandatoryExtensions = ['js', 'mjs', 'ts']

  options.extensions = mandatoryExtensions
    .filter(ext => !options.extensions.includes(ext))
    .concat(options.extensions)

  // If app.html is defined, set the template path to the user template
  if (options.appTemplatePath === undefined) {
    options.appTemplatePath = path.resolve(options.buildDir, 'views/app.template.html')
    if (fs.existsSync(path.join(options.srcDir, 'app.html'))) {
      options.appTemplatePath = path.join(options.srcDir, 'app.html')
    }
  } else {
    options.appTemplatePath = path.resolve(options.srcDir, options.appTemplatePath)
  }

  options.build.publicPath = options.build.publicPath.replace(/([^/])$/, '$1/')
  options.build._publicPath = options.build._publicPath.replace(/([^/])$/, '$1/')

  // Ignore publicPath on dev
  if (options.dev && isUrl(options.build.publicPath)) {
    options.build.publicPath = options.build._publicPath
  }

  // If store defined, update store options to true unless explicitly disabled
  if (
    options.store !== false &&
    fs.existsSync(path.join(options.srcDir, options.dir.store)) &&
    fs.readdirSync(path.join(options.srcDir, options.dir.store))
      .find(filename => filename !== 'README.md' && filename[0] !== '.')
  ) {
    options.store = true
  }

  // SPA loadingIndicator
  if (options.loadingIndicator) {
    // Normalize loadingIndicator
    if (!isPureObject(options.loadingIndicator)) {
      options.loadingIndicator = { name: options.loadingIndicator }
    }

    // Apply defaults
    options.loadingIndicator = Object.assign(
      {
        name: 'default',
        color: (options.loading && options.loading.color) || '#D3D3D3',
        color2: '#F5F5F5',
        background: (options.manifest && options.manifest.theme_color) || 'white',
        dev: options.dev,
        loading: options.messages.loading
      },
      options.loadingIndicator
    )
  }

  // Debug errors
  if (options.debug === undefined) {
    options.debug = options.dev
  }

  // Apply default hash to CSP option
  const { csp } = options.render

  const cspDefaults = {
    hashAlgorithm: 'sha256',
    allowedSources: undefined,
    policies: undefined,
    reportOnly: options.debug
  }
  if (csp) {
    options.render.csp = defaults(isObject(csp) ? csp : {}, cspDefaults)
  }

  // cssSourceMap
  if (options.build.cssSourceMap === undefined) {
    options.build.cssSourceMap = options.dev
  }

  const babelConfig = options.build.babel
  // babel cacheDirectory
  if (babelConfig.cacheDirectory === undefined) {
    babelConfig.cacheDirectory = options.dev
  }

  // TODO: remove this warn in Nuxt 3
  if (Array.isArray(babelConfig.presets)) {
    const warnPreset = (presetName) => {
      const oldPreset = '@nuxtjs/babel-preset-app'
      const newPreset = '@nuxt/babel-preset-app'
      if (presetName.includes(oldPreset)) {
        presetName = presetName.replace(oldPreset, newPreset)
        consola.warn('@nuxtjs/babel-preset-app has been deprecated, please use @nuxt/babel-preset-app.')
      }
      return presetName
    }
    babelConfig.presets = babelConfig.presets.map((preset) => {
      const hasOptions = Array.isArray(preset)
      if (hasOptions) {
        preset[0] = warnPreset(preset[0])
      } else if (typeof preset === 'string') {
        preset = warnPreset(preset)
      }
      return preset
    })
  }

  // vue config
  const vueConfig = options.vue.config

  if (vueConfig.silent === undefined) {
    vueConfig.silent = !options.dev
  }
  if (vueConfig.performance === undefined) {
    vueConfig.performance = options.dev
  }

  // merge custom env with variables
  const eligibleEnvVariables = pick(process.env, Object.keys(process.env).filter(k => k.startsWith('NUXT_ENV_')))
  Object.assign(options.env, eligibleEnvVariables)

  // Normalize ignore
  options.ignore = options.ignore ? [].concat(options.ignore) : []

  // Append ignorePrefix glob to ignore
  if (typeof options.ignorePrefix === 'string') {
    options.ignore.push(`**/${options.ignorePrefix}*.*`)
  }

  // Compression middleware legacy
  if (options.render.gzip) {
    consola.warn('render.gzip is deprecated and will be removed in a future version! Please switch to render.compressor')
    options.render.compressor = options.render.gzip
    delete options.render.gzip
  }

  // Apply mode preset
  const modePreset = options.modes[options.mode || 'universal']

  if (!modePreset) {
    consola.warn(`Unknown mode: ${options.mode}. Falling back to universal`)
  }
  defaultsDeep(options, modePreset || options.modes.universal)

  // If no server-side rendering, add appear true transition
  if (options.render.ssr === false && options.transition) {
    options.transition.appear = true
  }

  // We assume the SPA fallback path is 404.html (for GitHub Pages, Surge, etc.)
  if (options.generate.fallback === true) {
    options.generate.fallback = '404.html'
  }

  if (options.build.stats === 'none' || options.build.quiet === true) {
    options.build.stats = false
  }

  // Vendor backward compatibility with nuxt 1.x
  if (typeof options.build.vendor !== 'undefined') {
    delete options.build.vendor
    consola.warn('vendor has been deprecated due to webpack4 optimization')
  }

  // Disable CSS extraction due to incompatibility with thread-loader
  if (options.build.extractCSS && options.build.parallel) {
    options.build.parallel = false
    consola.warn('extractCSS cannot work with parallel build due to limited work pool in thread-loader')
  }

  // build.extractCSS.allChunks has no effect
  if (typeof options.build.extractCSS.allChunks !== 'undefined') {
    consola.warn('build.extractCSS.allChunks has no effect from v2.0.0. Please use build.optimization.splitChunks settings instead.')
  }

  // Enable minimize for production builds
  if (options.build.optimization.minimize === undefined) {
    options.build.optimization.minimize = !options.dev
  }

  // Enable optimizeCSS only when extractCSS is enabled
  if (options.build.optimizeCSS === undefined) {
    options.build.optimizeCSS = options.build.extractCSS ? {} : false
  }

  const { loaders } = options.build
  const vueLoader = loaders.vue
  if (vueLoader.productionMode === undefined) {
    vueLoader.productionMode = !options.dev
  }
  const styleLoaders = [
    'css', 'cssModules', 'less',
    'sass', 'scss', 'stylus', 'vueStyle'
  ]
  for (const name of styleLoaders) {
    const loader = loaders[name]
    if (loader && loader.sourceMap === undefined) {
      loader.sourceMap = Boolean(options.build.cssSourceMap)
    }
  }

  options.build.transpile = [].concat(options.build.transpile || [])

  if (options.build.quiet === true) {
    consola.level = 0
  }

  // Use runInNewContext for dev mode by default
  const { bundleRenderer } = options.render
  if (typeof bundleRenderer.runInNewContext === 'undefined') {
    bundleRenderer.runInNewContext = options.dev
  }

  return options
}
