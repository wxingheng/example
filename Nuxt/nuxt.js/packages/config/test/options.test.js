import path from 'path'
import consola from 'consola'
import { getNuxtConfig } from '../src/options'

jest.mock('std-env', () => ({
  browser: false,
  test: 'test',
  dev: false,
  production: true,
  debug: false,
  ci: true,
  windows: false,
  darwin: false,
  linux: true
}))

jest.mock('@nuxt/utils', () => ({
  ...jest.requireActual('@nuxt/utils'),
  getMainModule: () => ({ paths: ['/var/nuxt/node_modules'] })
}))

describe('config: options', () => {
  test('should return default nuxt config', () => {
    jest.spyOn(process, 'cwd').mockReturnValue('/var/nuxt/test')
    jest.spyOn(path, 'resolve').mockImplementation((...args) => args.join('/').replace(/\\+/, '/'))

    expect(getNuxtConfig({})).toMatchSnapshot()

    process.cwd.mockRestore()
    path.resolve.mockRestore()
  })

  test('should prevent duplicate calls with same options', () => {
    const options = {}
    const firstConfig = getNuxtConfig(options)
    const secondConfig = getNuxtConfig(firstConfig)

    expect(firstConfig).toBe(secondConfig)
    expect(firstConfig.__normalized__).toBe(true)
  })

  test('should return default loading when loading is true', () => {
    const { loading } = getNuxtConfig({ loading: true })
    expect(loading).toEqual({
      color: 'black',
      failedColor: 'red',
      height: '2px',
      throttle: 200,
      duration: 5000,
      continuous: false,
      rtl: false,
      css: true
    })
  })

  test('should transform transition/layoutTransition to name', () => {
    const { transition, layoutTransition } = getNuxtConfig({
      transition: 'test-tran',
      layoutTransition: 'test-layout-tran'
    })
    expect(transition).toMatchObject({ name: 'test-tran' })
    expect(layoutTransition).toMatchObject({ name: 'test-layout-tran' })
  })

  test('should transform extensions to array', () => {
    const { extensions } = getNuxtConfig({ extensions: 'ext' })
    expect(extensions).toEqual(['js', 'mjs', 'ts', 'ext'])
  })

  test('should support custom global name', () => {
    const { globalName } = getNuxtConfig({ globalName: 'globalNuxt' })
    expect(globalName).toEqual('globalnuxt')
  })

  test('should detect store dir', () => {
    const { store } = getNuxtConfig({ rootDir: path.resolve(__dirname, 'fixtures') })
    expect(store).toEqual(true)
  })

  test('should enable csp', () => {
    const { render: { csp } } = getNuxtConfig({ render: { csp: { allowedSources: true, test: true } } })
    expect(csp).toEqual({
      hashAlgorithm: 'sha256',
      allowedSources: true,
      policies: undefined,
      reportOnly: false,
      test: true
    })
  })

  test('should check unknown mode', () => {
    const { build, render } = getNuxtConfig({ mode: 'test' })
    expect(consola.warn).toHaveBeenCalledWith('Unknown mode: test. Falling back to universal')
    expect(build.ssr).toEqual(true)
    expect(render.ssr).toEqual(true)
  })

  test('should add appear true in transition when no ssr', () => {
    const { transition } = getNuxtConfig({ render: { ssr: false } })
    expect(transition.appear).toEqual(true)
  })

  test('should return 404.html as default generate.fallback', () => {
    const { generate: { fallback } } = getNuxtConfig({ generate: { fallback: true } })
    expect(fallback).toEqual('404.html')
  })

  test('should disable parallel if extractCSS is enabled', () => {
    const { build: { parallel } } = getNuxtConfig({ build: { extractCSS: true, parallel: true } })
    expect(parallel).toEqual(false)
    expect(consola.warn).toHaveBeenCalledWith('extractCSS cannot work with parallel build due to limited work pool in thread-loader')
  })

  describe('config: router dir', () => {
    test('should transform middleware to array', () => {
      const { router: { middleware } } = getNuxtConfig({ router: { middleware: 'midd' } })
      expect(middleware).toEqual(['midd'])
    })

    test('should set _routerBaseSpecified when base is specified', () => {
      const { _routerBaseSpecified } = getNuxtConfig({ router: { base: '/test' } })
      expect(_routerBaseSpecified).toEqual(true)
    })
  })

  describe('config: options dir', () => {
    test('should support custom root dir', () => {
      const { rootDir } = getNuxtConfig({
        rootDir: 'root'
      })
      expect(rootDir).toEqual(path.resolve('root'))
    })

    test('should support custom src dir', () => {
      const { srcDir } = getNuxtConfig({
        rootDir: 'root',
        srcDir: 'src'
      })
      expect(srcDir).toEqual(path.resolve('root', 'src'))
    })

    test('should support custom generate dir', () => {
      const { generate: { dir } } = getNuxtConfig({
        rootDir: 'root',
        generate: { dir: 'generate' }
      })
      expect(dir).toEqual(path.resolve('root', 'generate'))
    })
  })

  describe('config: options template', () => {
    test('should use default appTemplatePath', () => {
      const { appTemplatePath } = getNuxtConfig({})
      expect(appTemplatePath).toEqual(path.resolve('.nuxt', 'views', 'app.template.html'))
    })

    test('should use custom appTemplatePath', () => {
      const { appTemplatePath } = getNuxtConfig({ appTemplatePath: 'templates' })
      expect(appTemplatePath).toEqual(path.resolve('templates'))
    })

    test('should use custom app.html', () => {
      const { appTemplatePath } = getNuxtConfig({ rootDir: path.resolve(__dirname, 'fixtures') })
      expect(appTemplatePath).toEqual(path.resolve(__dirname, 'fixtures', 'app.html'))
    })
  })

  describe('config: options publicPath', () => {
    test('should fallback to default when publicPath is falsy', () => {
      const { build: { publicPath } } = getNuxtConfig({ build: { publicPath: false } })
      expect(publicPath).toEqual('/_nuxt/')
    })

    test('should append slash in publicPath', () => {
      const { build: { publicPath } } = getNuxtConfig({ build: { publicPath: '/nuxt_public' } })
      expect(publicPath).toEqual('/nuxt_public/')
    })

    test('should ignore url publicPath in dev', () => {
      const { build: { publicPath } } = getNuxtConfig({ dev: true, build: { publicPath: 'http://nuxt_public' } })
      expect(publicPath).toEqual('/_nuxt/')
    })
  })

  describe('config: options babel', () => {
    test('should replace and deprecate @nuxtjs/babel-preset-app', () => {
      const { build: { babel } } = getNuxtConfig({
        build: { babel: { presets: ['@nuxtjs/babel-preset-app'] } }
      })
      expect(consola.warn).toHaveBeenCalledWith('@nuxtjs/babel-preset-app has been deprecated, please use @nuxt/babel-preset-app.')
      expect(babel).toEqual({
        'babelrc': false,
        'cacheDirectory': false,
        'presets': ['@nuxt/babel-preset-app']
      })
    })

    test('should support options in babel presets', () => {
      const { build: { babel } } = getNuxtConfig({
        build: { babel: { presets: [['@nuxt/babel-preset-app', { test: true }]] } }
      })
      expect(babel).toEqual({
        'babelrc': false,
        'cacheDirectory': false,
        'presets': [['@nuxt/babel-preset-app', { test: true }]]
      })
    })
  })

  describe('config: options deprecated', () => {
    test('should deprecate render.gzip', () => {
      getNuxtConfig({ render: { gzip: true } })
      expect(consola.warn).toHaveBeenCalledWith('render.gzip is deprecated and will be removed in a future version! Please switch to render.compressor')
    })

    test('should deprecate build.vendor', () => {
      getNuxtConfig({ build: { vendor: ['lodash'] } })
      expect(consola.warn).toHaveBeenCalledWith('vendor has been deprecated due to webpack4 optimization')
    })

    test('should deprecate build.extractCSS.allChunks', () => {
      getNuxtConfig({ build: { extractCSS: { allChunks: true } } })
      expect(consola.warn).toHaveBeenCalledWith('build.extractCSS.allChunks has no effect from v2.0.0. Please use build.optimization.splitChunks settings instead.')
    })
  })
})
