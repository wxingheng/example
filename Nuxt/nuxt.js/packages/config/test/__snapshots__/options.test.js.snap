// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config: options should return default nuxt config 1`] = `
Object {
  "ErrorPage": null,
  "__normalized__": true,
  "_nuxtConfigFile": "/var/nuxt/test/nuxt.config.js",
  "appTemplatePath": "/var/nuxt/test/.nuxt/views/app.template.html",
  "build": Object {
    "_publicPath": "/_nuxt/",
    "analyze": false,
    "babel": Object {
      "babelrc": false,
      "cacheDirectory": false,
    },
    "cache": false,
    "crossorigin": undefined,
    "cssSourceMap": false,
    "devMiddleware": Object {},
    "extractCSS": false,
    "filenames": Object {
      "app": [Function],
      "chunk": [Function],
      "css": [Function],
      "font": [Function],
      "img": [Function],
      "video": [Function],
    },
    "friendlyErrors": true,
    "hardSource": false,
    "hotMiddleware": Object {},
    "html": Object {
      "minify": Object {
        "collapseBooleanAttributes": true,
        "decodeEntities": true,
        "minifyCSS": true,
        "minifyJS": true,
        "processConditionalComments": true,
        "removeEmptyAttributes": true,
        "removeRedundantAttributes": true,
        "trimCustomFragments": true,
        "useShortDoctype": true,
      },
    },
    "loaders": Object {
      "css": Object {
        "sourceMap": false,
      },
      "cssModules": Object {
        "localIdentName": "[local]_[hash:base64:5]",
        "sourceMap": false,
      },
      "file": Object {},
      "fontUrl": Object {
        "limit": 1000,
      },
      "imgUrl": Object {
        "limit": 1000,
      },
      "less": Object {
        "sourceMap": false,
      },
      "pugPlain": Object {},
      "sass": Object {
        "indentedSyntax": true,
        "sourceMap": false,
      },
      "scss": Object {
        "sourceMap": false,
      },
      "stylus": Object {
        "sourceMap": false,
      },
      "ts": Object {
        "appendTsSuffixTo": Array [
          /\\\\\\.vue\\$/,
        ],
        "transpileOnly": true,
      },
      "tsx": Object {
        "appendTsxSuffixTo": Array [
          /\\\\\\.vue\\$/,
        ],
        "transpileOnly": true,
      },
      "vue": Object {
        "productionMode": true,
        "transformAssetUrls": Object {
          "embed": "src",
          "object": "src",
          "source": "src",
          "video": "src",
        },
      },
      "vueStyle": Object {
        "sourceMap": false,
      },
    },
    "optimization": Object {
      "minimize": true,
      "minimizer": undefined,
      "runtimeChunk": "single",
      "splitChunks": Object {
        "automaticNameDelimiter": ".",
        "cacheGroups": Object {},
        "chunks": "all",
        "name": undefined,
      },
    },
    "optimizeCSS": false,
    "parallel": false,
    "plugins": Array [],
    "postcss": Object {
      "preset": Object {
        "stage": 2,
      },
    },
    "profile": false,
    "publicPath": "/_nuxt/",
    "quiet": true,
    "splitChunks": Object {
      "commons": true,
      "layouts": false,
      "pages": true,
    },
    "ssr": true,
    "standalone": false,
    "stats": false,
    "styleResources": Object {},
    "template": undefined,
    "templates": Array [],
    "terser": Object {},
    "transpile": Array [],
    "typescript": Object {
      "typeCheck": true,
    },
    "watch": Array [],
  },
  "buildDir": "/var/nuxt/test/.nuxt",
  "cli": Object {
    "badgeMessages": Array [],
  },
  "css": Array [],
  "debug": false,
  "dev": false,
  "devModules": Array [],
  "dir": Object {
    "assets": "assets",
    "layouts": "layouts",
    "middleware": "middleware",
    "pages": "pages",
    "static": "static",
    "store": "store",
  },
  "editor": undefined,
  "env": Object {},
  "extensions": Array [
    "js",
    "mjs",
    "ts",
  ],
  "fetch": Object {
    "client": true,
    "server": true,
  },
  "generate": Object {
    "concurrency": 500,
    "dir": "/var/nuxt/test/dist",
    "exclude": Array [],
    "fallback": "200.html",
    "interval": 0,
    "routes": Array [],
    "subFolders": true,
  },
  "globalName": "nuxt",
  "globals": Object {
    "context": [Function],
    "id": [Function],
    "loadedCallback": [Function],
    "nuxt": [Function],
    "pluginPrefix": [Function],
    "readyCallback": [Function],
  },
  "head": Object {
    "link": Array [],
    "meta": Array [],
    "script": Array [],
    "style": Array [],
  },
  "hooks": null,
  "ignore": Array [
    "**/*.test.*",
    "**/*.spec.*",
    "**/-*.*",
  ],
  "ignorePrefix": "-",
  "layoutTransition": Object {
    "mode": "out-in",
    "name": "layout",
  },
  "layouts": Object {},
  "loading": Object {
    "color": "black",
    "continuous": false,
    "css": true,
    "duration": 5000,
    "failedColor": "red",
    "height": "2px",
    "rtl": false,
    "throttle": 200,
  },
  "loadingIndicator": Object {
    "background": "white",
    "color": "black",
    "color2": "#F5F5F5",
    "dev": false,
    "loading": "Loading...",
    "name": "default",
  },
  "messages": Object {
    "back_to_home": "Back to the home page",
    "client_error": "Error",
    "client_error_details": "An error occurred while rendering the page. Check developer tools console for details.",
    "error_404": "This page could not be found",
    "loading": "Loading...",
    "nuxtjs": "Nuxt.js",
    "server_error": "Server error",
    "server_error_details": "An error occurred in the application and your page could not be served. If you are the application owner, check your logs for details.",
  },
  "mode": "universal",
  "modern": undefined,
  "modes": Object {
    "spa": Object {
      "build": Object {
        "ssr": false,
      },
      "render": Object {
        "ssr": false,
      },
    },
    "universal": Object {
      "build": Object {
        "ssr": true,
      },
      "render": Object {
        "ssr": true,
      },
    },
  },
  "modules": Array [],
  "modulesDir": Array [
    "/var/nuxt/node_modules",
    "/var/nuxt/test/node_modules",
  ],
  "plugins": Array [],
  "render": Object {
    "bundleRenderer": Object {
      "runInNewContext": false,
      "shouldPrefetch": [Function],
      "shouldPreload": [Function],
    },
    "compressor": Object {
      "threshold": 0,
    },
    "csp": false,
    "dist": Object {
      "index": false,
      "maxAge": "1y",
    },
    "etag": Object {
      "weak": false,
    },
    "fallback": Object {
      "dist": Object {},
      "static": Object {
        "handlers": Object {
          ".htm": false,
          ".html": false,
        },
        "skipUnknown": true,
      },
    },
    "http2": Object {
      "push": false,
      "pushAssets": null,
      "shouldPush": null,
    },
    "resourceHints": true,
    "ssr": true,
    "static": Object {
      "prefix": true,
    },
  },
  "rootDir": "/var/nuxt/test",
  "router": Object {
    "base": "/",
    "extendRoutes": null,
    "fallback": false,
    "linkActiveClass": "nuxt-link-active",
    "linkExactActiveClass": "nuxt-link-exact-active",
    "linkPrefetchedClass": false,
    "middleware": Array [],
    "mode": "history",
    "parseQuery": false,
    "prefetchLinks": true,
    "routeNameSplitter": "-",
    "routes": Array [],
    "scrollBehavior": null,
    "stringifyQuery": false,
  },
  "server": Object {
    "host": "localhost",
    "https": false,
    "port": 3000,
    "socket": undefined,
    "timing": false,
  },
  "serverMiddleware": Array [],
  "srcDir": "/var/nuxt/test",
  "styleExtensions": Array [
    "css",
    "pcss",
    "postcss",
    "styl",
    "stylus",
    "scss",
    "sass",
    "less",
  ],
  "test": true,
  "transition": Object {
    "appear": false,
    "appearActiveClass": "appear-active",
    "appearClass": "appear",
    "appearToClass": "appear-to",
    "mode": "out-in",
    "name": "page",
  },
  "vue": Object {
    "config": Object {
      "performance": false,
      "silent": true,
    },
  },
  "watch": Array [
    "/var/nuxt/test/nuxt.config.js",
  ],
  "watchers": Object {
    "chokidar": Object {
      "ignoreInitial": true,
    },
    "rewatchOnRawEvents": Array [
      "rename",
    ],
    "webpack": Object {},
  },
}
`;
